File: Pro_ODP_NET_for_Oracle_Database_11g-4541/vb_dot_net/Chapter9VBNET/UDT/OrderFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Oracle.DataAccess.Client
Imports Oracle.DataAccess.Types

'JobClass factory class
<OracleCustomTypeMappingAttribute("EDZEHOO.JOBS_TYPE")> _
Public Class OrderFactory
    Implements IOracleCustomTypeFactory

    ' Implementation of IOracleCustomTypeFactory.CreateObject()
    Public Function CreateObject() As IOracleCustomType Implements Oracle.DataAccess.Types.IOracleCustomTypeFactory.CreateObject
        Return New JobClass()
    End Function 'CreateObject

End Class 'OrderFactory

File: Pro_ODP_NET_for_Oracle_Database_11g-4541/c_sharp/Chapter 9/UDT/OrderFactory.cs
using System;
using System.Collections.Generic;
using System.Text;
using Oracle.DataAccess.Client;
using Oracle.DataAccess.Types;

//JobClass factory class
[OracleCustomTypeMappingAttribute("EDZEHOO.JOBS_TYPE")]
public class OrderFactory : IOracleCustomTypeFactory
{
    // Implementation of IOracleCustomTypeFactory.CreateObject()
    public IOracleCustomType CreateObject()
    {
        return new JobClass();
    }
}